### ISSUE TITLE
HTTP Header Parsing is wrong in 0.11.1

### ISSUE THREAD

**Issue by speedpat** [cpp-netlib/cpp-netlib#487]
in boost/network/protocol/http/client/connection/sync_base.hpp 
in the 
void read_headers(...) 
function it is assumed that the HTTP header field content starts 1 byte after the colon.
(colon_offset + 2)

response_ << header(name, header_line.substr(colon_offset + 2));

From the HTTP/1.1 Spec (http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2)
[...]
The field value MAY be preceded by any amount of LWS, though a single SP is preferred.
[...]

**Comment by deanberris** [comment]
Interesting. I agree this seems broken. Can you send in a pull request to
fix this?

**Comment by speedpat** [comment]
sure, I'm working on a testcase for it.

**Comment by deanberris** [comment]
Actually this looks like it's a real bug. re-opening.

**Comment by speedpat** [comment]
I already sent a pull request. But I'm struggling with getting the test to work with travis.

**Comment by glynos** [comment]
Maybe it[s something we can help with? What is the issue you're having with Travis?

**Comment by speedpat** [comment]
in pull request #490 I have a testcase. executing it local on my machine succeeds. but travis hung on executing it.

**Comment by deanberris** [comment]
Right. So as I mentioned in #490 and #495 this is being done on the synchronous client, which is no longer in 0.12. In the asynchronous client, this has to be done in the incremental parser instead, which is testable in isolation. Do you want to try and do that instead @speedpat?

**Comment by speedpat** [comment]
Sure, I'll do it. @deanberris

**Comment by deanberris** [comment]
Any update on this @speedpat -- I've just tagged an rc1 for 0.12.0 and would love to get this fixed in the next RC.
